src/actions/menu_items_actions.ts

```diff
@@ -55,12 +56,13 @@ async function paste(env: SpreadsheetChildEnv, pasteOption?: ClipboardPasteOptio
   const osClipboard = await env.clipboard.read();
   switch (osClipboard.status) {
     case "ok":
-      const clipboardContent = parseOSClipboardContent(osClipboard.content);
-      const clipboardId = clipboardContent.data?.clipboardId;
+      const clipboardId = env.model.getters.getClipboardId();
+      const clipboardContent = await parseOSClipboardContent(env, osClipboard.content, clipboardId);
+      const contentClipboardId = clipboardContent.data?.clipboardId;
```

Comment:
```suggestion
      const osClipboardContent = await parseOSClipboardContent(env, osClipboard.content, clipboardId);
      const osClipboardId = clipboardContent.data?.clipboardId;
```
nitpick. Mayeb those are better names ? Same in grid